app/0.14.x/loadbalancing.md

```diff
@@ -91,14 +94,16 @@ with 527 entries, whereas weights 16 and 32 (or their smallest relative
 counterparts 1 and 2) would result in a structure with merely 3 entries,
 especially with a very small (or even 0) `ttl` value.
 
-- Some nameservers do not return all entries (due to UDP packet size) in those
-cases (for example Consul returns a maximum of 3) a given Kong node will only
+- Some nameservers do not return all entries due to UDP packet size
+(for example Consul returns a maximum of 3 DNS entries). In those
```

Comment:
The Consul DNS server (and possibly other servers) returns a maximum of three values for a particular record via conventional UDP query to prevent returning a truncation flag. It returns all values for a particular record via EDNS UDP query or TCP query. Conventional DNS clients/libraries (without support for EDNS) such as `lua-resty-dns` in Kong only revert to a TCP query if the UDP response includes a truncation flag. Thus, Kong never reverts to a TCP query in environments using Consul DNS and may not consistently route to all instances of a service if more than three instances exist.

Reply:
I think the problem here is with Consul, which tries to be smart and prevents returning the truncation flag. It shouldn't do this, or at least have the behaviour configurable.

Reply:
From the Consul documentation...

> When the DNS query is performed using UDP, Consul will truncate the results without setting the truncate bit. This is to prevent a redundant lookup over TCP that generates additional load. If the lookup is done over TCP, the results are not truncated.

However, one can configure Consul to use the truncate flag via the `enable_truncate` option. I would still prefer a way to configure Kong to always use TCP queries if the resolver library cannot support EDNS.